base/applications/utilman/process.c

```diff
@@ -0,0 +1,178 @@
+    return bIsRunning;
+}
+
+/**
+ * @LaunchProcess
+ *
+ * Executes a process.
+ *
+ * @param[in]   lpProcessName
+ *     The name of the executable process.
+ *
+ * @return
+ *     Returns TRUE if the process has been launched successfully,
+ *     FALSE otherwise.
+ *
+ */
+BOOL LaunchProcess(LPCWSTR lpProcessName)
+{
+    STARTUPINFOW si;
+    PROCESS_INFORMATION pi;
+    INT_PTR Success;
+    WCHAR ExpandedCmdLine[MAX_PATH];
+
+    ExpandEnvironmentStringsW(lpProcessName, ExpandedCmdLine, ARRAYSIZE(ExpandedCmdLine));
+
+    ZeroMemory(&pi, sizeof(pi));
+    ZeroMemory(&si, sizeof(si));
+    si.cb = sizeof(si);
+    si.dwFlags = STARTF_USESHOWWINDOW;
+    si.wShowWindow = SW_SHOWNORMAL;
+
+    if (!CreateProcessW(NULL,
+                        ExpandedCmdLine,
+                        NULL,
+                        NULL,
+                        FALSE,
+                        0, // DETACHED_PROCESS, // NORMAL_PRIORITY_CLASS,
+                        NULL,
+                        NULL,
+                        &si,
+                        &pi))
+    {
+        DPRINT("CreateProcessW() failed with error %lu\n", GetLastError());
+
+        /*
+         * Windows Vista and above require elevated privileges. Instead of
+         * CreateProcessW() we'll use ShellExecuteW() to launch the application as the function
+         * is aware of what elevation should be prompted.
+        */
+        Success = (INT_PTR)ShellExecuteW(NULL, L"open", ExpandedCmdLine, NULL, NULL, SW_SHOWNORMAL);
+
+        /*
+         * ShellExecuteW() returns a value that is greater than 32 indicating the
+         * function has succeded. If the function fails, the reason of such failure
+         * is shown by GetLastError().
+         */
+        if (Success >= 32)
+            return TRUE;
+
+        return FALSE;
+    }
+
+    CloseHandle(pi.hProcess);
+    CloseHandle(pi.hThread);
+    return TRUE;
+}
+
+/**
+ * @CloseProcess
+ *
+ * Closes a process.
+ *
+ * @param[in]   lpProcessName
+ *     The name of the executable process.
+ *
+ * @return
+ *     Returns TRUE if the process has been terminated successfully,
+ *     FALSE otherwise.
+ *
+ */
+BOOL CloseProcess(IN LPCWSTR lpProcessName)
+{
+    BOOL bSuccess = FALSE;
+    PROCESSENTRY32W Process = {0};
+
+    /* Create a snapshot and check if the given process' executable name is running */
+    HANDLE hSnapshot = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
+
+    if (hSnapshot == INVALID_HANDLE_VALUE)
+        return FALSE;
+
+    Process.dwSize = sizeof(Process);
+
+    /* Enumerate the processes */
+    if (Process32FirstW(hSnapshot, &Process))
+    {
+        do
+        {
+            if (_wcsicmp(Process.szExeFile, lpProcessName) == 0)
+            {
+                /*
+                 * We have found the process. Retrieve its process ID
+                 * and check that the handle is not NULL. If the ID
+                 * matches with the specific process then terminate it.
```

Comment:
`If the ID matches with the specific process then terminate it.`
That is not really what is being checked here,
the check is that we do not kill ourself...